### PR TITLE
[JENKINS-36143] Allow rc classifier

### PR DESCRIPTION
**Pull request by andresrc**
See [JENKINS-36143](https://issues.jenkins-ci.org/browse/JENKINS-36143)

@reviewbybees

### CODE REVIEW

File: setup.mk [hunk 1]
```diff
@@ -35,4 +35,3 @@ export LICENSE_TEXT_COMMENTED:=$(shell echo "$(LICENSE_TEXT_COLUMN)" | sed  's!^
 
 # Put a dot in place of an empty line, and prepend a space
 export LICENSE_TEXT_DEB:=$(shell echo "$(LICENSE_TEXT_COLUMN)" | sed -e 's!^$$!.!g' -e 's!^! !g' )
-
```
Comment (jtnord) [inline setup.mk]: I thought keeping the trailing line is a good thing on Unix :-)

**Comment by jtnord** [comment]
I'm not so sure.  What happens to the [Jenkins LTS RCs](http://mirrors.jenkins-ci.org/war-stable-rc/1.651.3/) [deb link](http://pkg.jenkins-ci.org/debian-stable-rc/)?  

I think this is OK as the Jenkins OSS seems not to be using an RC release, but a SNAPSHOT that gets published - but would like to hear some confirmation from @olivergondza before this is merged.

File: setup.mk (hunk 1 above)
Comment (andresrc) [inline setup.mk]: Ups, I have to review the editor settings

**Comment by andresrc** [comment]
@jtnord AFAIU that part comes from the branding. Version numbers seem to be always stripped of "decorations".

**Comment by olivergondza** [comment]
We use `war.publish` goal to push the LTS RC and it would break it (as I understand the change). It would push it to non-`rc` directory.

**Comment by andresrc** [comment]
The WAR directory comes from the branding and environment:
* https://github.com/jenkinsci/packaging/blob/0efc63b24b5d0807b66657c5ad61dc8d1f58af15/env/release.mk#L12
* https://github.com/jenkinsci/packaging/blob/0efc63b24b5d0807b66657c5ad61dc8d1f58af15/env/test.mk#L13

Version number decorators are stripped in the RC from the version number as well:
* http://mirrors.jenkins-ci.org/war-rc/
* http://mirrors.jenkins-ci.org/war-stable-rc/

**Comment by jtnord** [comment]
@andresrc I'm pretty sure you are correct on the analysis.  the -rc lines define the location and that is specified as options to MAKE not retreived from the version.

**Comment by jtnord** [comment]
:bee:

**Comment by svanoort** [comment]
🐝

**Comment by olivergondza** [comment]
@andresrc, @jtnord, you are right. :bee:

**Pull request merged**